rgw/admin/mock.go

```diff
@@ -0,0 +1,18 @@
+package admin
+
+import (
+	"net/http"
+)
+
+// MockClient is the mock of the HTTP Client
+// It can be used to mock HTTP request/response from the rgw admin ops API
+type MockClient struct {
```

Comment:
I don't see a reason to export this, since the tests are run from inside the admin package.

Reply:
What about consumers of the library? Should they write their own `MockClient`?

Reply:
Well, they have to anyways, right? What we have here is just a wrapper around a function pointer. The function itself must be implemented in any case. The problem I see is that if you export something, it becomes part of the contract with the consumers, and you can't change it anymore. That's exactly what the `HTTPClient` interface is for in this case. And "writing" a `MockClient` is as easy as
```go
type MockDo func(req *http.Request) (*http.Response, error)

func (do MockDo) Do(req *http.Request) (*http.Response, error) { return do(req) }
```
So, I think it's good (for a library) to practice export hygiene and only export what is necessary. I actually would even move this code into the test file.

Reply:
Refernce: https://github.com/golang/go/wiki/CodeReviewComments#interfaces

Reply:
https://github.com/ceph/go-ceph/pull/539